client/common/directives/ModelFieldReference/ModelFieldReference.js

```diff
@@ -139,6 +146,24 @@ angular.module('dashboard.directives.ModelFieldReference', [
         GeneralModelService.list(apiPath, params, {preventCancel: true}).then(function(response) {
           if (!response) return; //in case http request was cancelled by newer request
           scope.list = response;
+          /**
+           * If option removeSelected is true
+           * the shared variable of model-field-reference-edit (modelReferenceEditData.items)
+           * will be used to save selected data for the comparison. This was tested only on
+           * single selection and schedule templates. Only available in single selection.
+           * It removes data that is in scope.list base 
+           * on the modelReferenceEditData.items
+           * @param  {Boolean} scope.options.removeSelected true or false
```

Comment:
I appreciate your effort in documenting this, but there are tradeoffs that we need to consider. Let's put priority in readability: this means we want to be able to read a function's code without having to scroll. Let's limit our comments to a few lines. Thanks